### PR TITLE
Add interval to PercentilePruner.

### PR DESCRIPTION
**Pull request by keisuke-umezawa**
## Motivation

Address #1136 . And to take over #1234.

## Description of the changes

1. Add `is_target_step` to `BasePruner` and implement `is_target_step` in `PercentilePruner`.
2. Use `pruner.is_target_step(epoch, trial)` in `KerasPruningCallback` to skip pruning by interval.

### CODE REVIEW

File: optuna/pruners/_percentile.py [hunk 1]
```diff
@@ -199,3 +194,16 @@ def prune(self, study: "optuna.study.Study", trial: "optuna.trial.FrozenTrial")
         if direction == StudyDirection.MAXIMIZE:
             return best_intermediate_result < p
         return best_intermediate_result > p
+
+    def is_target_step(self, step: int, trial: "optuna.trial.FrozenTrial") -> bool:
```
Comment (keisuke-umezawa) [inline optuna/pruners/_percentile.py:198]: To call `_is_first_in_interval_step` in this function, I need to add `trial` as an argument.

**Comment by keisuke-umezawa** [comment]
Blocked by this pr. https://github.com/optuna/optuna/pull/2171

**Comment by keisuke-umezawa** [comment]
@toshihikoyanase 
The blocker of this PR is merged. PTAL.

**Comment by Crissman** [comment]
@keisuke-umezawa please take a look at the conflict. 🙇‍♂️

File: optuna/pruners/_percentile.py (hunk 1 above)
Comment (keisuke-umezawa) [reply optuna/pruners/_percentile.py:198]: is_prunable_step should be better.

**Comment by keisuke-umezawa** [comment]
I reached the conclusion that this implementation is not what users want. Instead of it, I opend https://github.com/optuna/optuna/pull/2490.

**Pull request closed without merging**